### PR TITLE
doc: removed useless information for easier maintainence

### PR DESCRIPTION
**Pull request by hermet**
That basic information is addressed in the Results section.
We can only add information that pertains to a specific scenario.

### CODE REVIEW

**Comment by mgrudzinska** [comment]
Zrzut ekranu 2024-06-24 o 11 20 34

@hermet isn't this a bit confusing and looks like the fun always produces some issue?

**Comment by hermet** [comment]
> Zrzut ekranu 2024-06-24 o 11 20 34
> @hermet isn't this a bit confusing and looks like the fun always produces some issue?

@mgrudzinska could you please make an issue if it is not related to this PR?

**Comment by mgrudzinska** [comment]
> @mgrudzinska could you please make an issue if it is not related to this PR?

but it is related. I've suggested that if we remove all '@retval Result::Success ...` it will be confusing.
if you don't agree, please resolve my comment, no need to discuss

**Comment by hermet** [comment]
@mgrudzinska Fixing the return values of per API would easily harm the compatibility between the app and the library(You've already made this by changing the retval list of some apis). The reason we have listed the return values in enum, is to indicate that all APIs can potentially return these values. Therefore, users can compare the returned values to identify the type of error. This flexible approach ultimately ensures compatibility and easy to use.

```
enum class Result
{
    Success = 0,           ///< The value returned in case of a correct request execution.
    InvalidArguments,      ///< The value returned in the event of a problem with the arguments given to the API - e.g. empty paths or null pointers.
    InsufficientCondition, ///< The value returned in case the request cannot be processed - e.g. asking for properties of an object, which does not exist.
    FailedAllocation,      ///< The value returned in case of unsuccessful memory allocation.
    MemoryCorruption,      ///< The value returned in the event of bad memory handling - e.g. failing in pointer releasing or casting
    NonSupport,            ///< The value returned in case of choosing unsupported engine features(options).
    Unknown                ///< The value returned in all other cases.
};
```
This is not only about ThorVG's mechanism; many libraries, since the Unix system, follow this approach. Now, we can address the retval only when we can help users understand the reason for the non-success in more detail than the general explanation.

**Comment by hermet** [comment]
> if we remove all '@RetVal Result::Success ...` it will be confusing.

> > @mgrudzinska could you please make an issue if it is not related to this PR?
> 
> but it is related. I've suggested that if we remove all '@RetVal Result::Success ...` it will be confusing. if you don't agree, please resolve my comment, no need to discuss

@mgrudzinska It needs a logic. If you think my logic does not make sense, it is better to discuss it rather than ignore it. Please come up with a logic for what will happen to users if every API doesn't mention: "@RetVal Result::Success." I don't anticipate any confusion there.

File: inc/thorvg.h [hunk 1]
```diff
@@ -72,6 +72,10 @@ class Animation;
 
 /**
  * @brief Enumeration specifying the result from the APIs.
+ *
+ * All ThorVG APIs could potentially return one of the values in the list.
+ * Please note that some APIs may additionally specify the reasons that trigger their return values.
```
Comment (hermet) [inline inc/thorvg.h:77]: @mgrudzinska please note this

**Pull request merged**